shap/benchmark/framework.py

```diff
@@ -0,0 +1,110 @@
+import matplotlib.pyplot as plt 
+import pandas as pd 
+import numpy as np 
+import sklearn 
+import shap 
+from sklearn.model_selection import train_test_split
+from shap.utils import safe_isinstance, MaskedModel
+from shap.benchmark import SequentialPerturbation
+
+def update(model, X, y, explainer, masker, sort_order, score_function, perturbation, scores):
+    metric = perturbation + ' ' + sort_order
+    sp = SequentialPerturbation(model, masker, sort_order, score_function, perturbation)
+    x, y, auc = sp.score(explainer, X, y=y)
+    scores['metrics'].append(metric)
+    scores['values'][metric] = [x, y, auc] 
+
+def get_benchmark(model, X, y, explainer, masker, metrics, exp_num=1, *args):
+    # convert dataframes
+    if safe_isinstance(X, "pandas.core.series.Series") or safe_isinstance(X, "pandas.core.frame.DataFrame"):
```

Comment:
I think these kinds on strings should be moved to one constants file:
"pandas.core.series.Series"
this will make sure:
1.) if we write it incorrectly it will be easier to catch since it will be same everywhere
2.) if we need to update it we only need to update in one place

Reply:
Will do that in another PR.